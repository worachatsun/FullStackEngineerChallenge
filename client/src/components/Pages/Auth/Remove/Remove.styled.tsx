import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  padding: 30px;
`;

export const ButtonContainer = styled.div`
  margin-top: 20px;
  display: flex;
  justify-content: flex-end;
`;

export const Lable = styled.div`
  font-size: 1.5em;
  margin-bottom: 20px;
`;
