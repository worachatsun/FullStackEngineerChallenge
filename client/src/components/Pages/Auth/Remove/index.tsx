import React, { FunctionComponent } from "react";
import { mutate } from "swr";
import { LIST_EMPLOYEES_API, REMOVE_API } from "../../../../constants/routes";
import Button from "../../../commons/Button";
import Modal from "../../../commons/Modal";
import { HttpMethod, mutator } from "../../../commons/utils/client";
import useModal from "../../../Hook/useModal";
import { ButtonContainer, Container, Lable } from "./Remove.styled";

interface IProps {
    username: string;
    mutate?: () => void;
}

const Remove: FunctionComponent<IProps> = ({ username }) => {
    const { isShowing, toggle, close } = useModal();

    const onSubmit = async () => {
        const token = localStorage.getItem("token") || "";
        try {
            const { response } = await mutator(REMOVE_API, HttpMethod.POST, token, {
                username,
            });
            if (!response?.ok) throw new Error(`${response?.status}: ${response?.statusText}`);
            const data = await response?.json();
            if (data.error) throw new Error(`Unable to remove user`);
            mutate([LIST_EMPLOYEES_API, token]);
            close();
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <>
            <Button onClick={toggle}>Remove</Button>
            <Modal isShowing={isShowing} close={close}>
                <Container>
                    <Lable>Are you sure you want to delete this account?</Lable>
                    <ButtonContainer>
                        <Button onClick={onSubmit}>Remove</Button>
                    </ButtonContainer>
                </Container>
            </Modal>
        </>
    );
};

export default Remove;
