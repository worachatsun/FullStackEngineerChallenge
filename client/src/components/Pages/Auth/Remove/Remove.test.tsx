import React from "react";
import renderer from "react-test-renderer";
import Remove from "./index";

describe("<Remove />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<Remove username={"test"} mutate={jest.fn} />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
