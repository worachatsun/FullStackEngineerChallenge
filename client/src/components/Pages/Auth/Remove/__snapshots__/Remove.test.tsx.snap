// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Remove /> should render correctly 1`] = `
<button
  className="sc-AxjAm fbqdBC"
  onClick={[Function]}
>
  Remove
</button>
`;
